File: src/main/java/expensereport/Expense.java
package expensereport;

import static expensereport.Expense.Type.BREAKFAST;
import static expensereport.Expense.Type.DINNER;

public class Expense {
    boolean isOverage() {
        return (type == DINNER && amount > 5000)
                || (type == BREAKFAST && amount > 1000);
    }

    String getName() {
        String name = "TILT";
        switch (type) {
            case DINNER:
                name = "Dinner";
                break;
            case BREAKFAST:
                name = "Breakfast";
                break;
            case CAR_RENTAL:
                name = "Car Rental";
                break;
        }
        return name;
    }

    public enum Type {DINNER, BREAKFAST, CAR_RENTAL}

    ;
    public Type type;
    public int amount;

    public Expense(Type type, int amount) {
        this.type = type;
        this.amount = amount;
    }
}

File: SwiftExpense/SwiftExpense/Example/Expense.swift
//
//  Expense.swift
//  SwiftExpense
//
//  Created by tskim on 2019/11/06.
//  Copyright © 2019 assin. All rights reserved.
//

class Expense {
    
    enum ExpenseType {
        case dinner
        case breakfast
        case carRental
    }
    
    let type: ExpenseType
    let amount: Int

    public init(type: ExpenseType, amount: Int) {
        self.type = type
        self.amount = amount
    }
}
